src/components/Textfield.tsx

```diff
@@ -0,0 +1,75 @@
+import { InputHTMLAttributes, useState } from "react";
+import "./css/InputCommon.css";
+import "./css/Textfield.css";
+import passwordIconOff from "../img/btn_visibility_off.svg";
+import passwordIconOn from "../img/btn_visibility_on.svg";
+
+/*
+  [Textfield 필수 속성]
+  - onValueChange: 컴포넌트 밖에서 텍스트필드 값을 변경하는 메소드 전달 필요
+  
+  [Textfield 상태 속성]
+  - isValid: 에러 상태 표시 => null(초기상태). true 값을 넘기면 메시지 출력
+  - message: 메세지가 있는 경우 출력
+*/
+
+interface TextfieldProps extends InputHTMLAttributes<HTMLInputElement> {
+  onValueChange?: (name: string, value: string) => void;
+  isValid?: boolean | null;
+  message?: string;
+  className?: string;
+  type?: string;
+}
+
+const Textfield = ({
+  onValueChange = () => {},
+  isValid = null,
+  message,
+  className,
+  type,
+  ...rest
+}: TextfieldProps) => {
+  const statusMessageClass: Record<string, string> = {
+    null: "",
+    false: "input__message--error",
+  };
+
+  const hasMessage = !!message;
+  const isInvalid = isValid === false;
+  const showMessage = hasMessage && isInvalid;
+  const [showPassword, setShowPassword] = useState(false);
+
+  const onTogglePassword = () => {
+    setShowPassword((prev) => !prev);
+  };
+
+  return (
+    <div className="input__container">
+      <div className="input__wrapper">
+        <input
+          onChange={(e) => onValueChange(e.target.name, e.target.value)}
+          className={`input textfield ${className}`}
+          {...rest}
+        />
+        {type === "password" && (
+          <img
+            src={showPassword ? passwordIconOn : passwordIconOff}
+            width="24px"
+            className="input__password__icon"
+            aria-label="비밀번호 표시"
+            aria-pressed="false"
+            role="button"
+            onClick={onTogglePassword}
+          />
+        )}
+      </div>
+      {showMessage && (
+        <div className={`${statusMessageClass[String(isValid)]}`}>
```

Comment:
isValid가 null일 때 String(null)은 "null"이 되어 statusMessageClass의 키로 사용되는데, 타입 안정성이 떨어지고 예측하기 어렵게 만드는 문제가 있네요!

부정확한 타입 정의를 제거하고 코드 안정성을 높이기위해 statusMessageClass mapper를 만드는대신, 아래와 같이 타입이 정의된 렌더러 함수를 사용해보는 방법은 어떨까요?

```tsx
  const getErrorMessageClass = (isValid: boolean | null): string => {
    if (isValid === false) {
      return "input__message--error";
    }
    return "";
  };
```
```suggestion
        <div className={getErrorMessageClass(isValid)}>{message}</div>
```

Reply:
타입 안정성이 떨어지는 문제가 있었네요! 함수로 변경하여 적용하겠습니다